coast/NEMO.py

```diff
@@ -278,6 +279,129 @@ def transect_indices(self, start: tuple, end: tuple) -> tuple:
+        
+        Returns
+        -------
+        Array of x indices, Array of y indices
+        '''
+        # Cast lat/lon to numpy arrays in case xarray things
+        new_lons = np.array(new_lons)
+        new_lats = np.array(new_lats)
+        mod_lon = np.array(model_dataset.longitude).flatten()
+        mod_lat = np.array(model_dataset.latitude).flatten()
+        
+        # Put lons and lats into 2D location arrays for BallTree: [lat, lon]
+        mod_locs = np.vstack((mod_lat, mod_lon)).transpose()
+        new_locs = np.vstack((new_lats, new_lons)).transpose()
+        
+        # Convert lat/lon to radians for BallTree
+        mod_locs = np.radians(mod_locs)
+        new_locs = np.radians(new_locs)
+        
+        # Do nearest neighbour interpolation using BallTree (gets indices)
+        tree = nb.BallTree(mod_locs, leaf_size=5, metric='haversine')
+        _, ind_1d = tree.query(new_locs, k=1)
+        
+        # Get 2D indices from 1D index output from BallTree
+        ind_y, ind_x = np.unravel_index(ind_1d, model_dataset.longitude.shape)
+        ind_x = xr.DataArray(ind_x.squeeze())
+        ind_y = xr.DataArray(ind_y.squeeze())
+
+        return ind_x, ind_y
+    
+    def nearest_time_indices(self):
+        raise NotImplementedError
+        return
+    
+    def nearest_depth_indices(self):
+        raise NotImplementedError
+        return
+    
+    def interpolate_in_space(self, model_array, new_lons, new_lats):
+        '''
+        Interpolates a provided xarray.DataArray in space to new longitudes
+        and latitudes using a nearest neighbour method (BallTree).
+        
+        Example Useage
+        ----------
+        # Get an interpolated DataArray for temperature onto two locations
+        interpolated = nemo.interpolate_in_space(nemo.dataset.votemper,
+                                                 [0,1], [45,46])
+        Parameters
+        ----------
+        model_array (xr.DataArray): Model variable DataArray to interpolate
+        new_lons (1Darray): Array of longitudes (degrees) to compare with model
+        new_lats (1Darray): Array of latitudes (degrees) to compare with model
+        
+        Returns
+        -------
+        Interpolated DataArray
+        '''
+        
+        # Get nearest indices
+        ind_x, ind_y = self.nearest_xy_indices(model_array, new_lons, new_lats)
+        
+        # Geographical interpolation (using BallTree indices)
+        interpolated = model_array.isel(x_dim=ind_x, y_dim=ind_y)
+        interpolated = interpolated.rename({'dim_0':'interp_dim'})
+        return interpolated
+    
+    def interpolate_in_time(self, model_array, new_times, 
+                               interp_method = 'nearest', extrapolate=True):
+        '''
+        Interpolates a provided xarray.DataArray in time to new python
+        datetimes using a specified scipy.interpolate method.
+        
+        Example Useage
+        ----------
+
```

Comment:
Add something to Example Usage. This sort of thing is really worth the effort. Then add to unit test

Reply:
@jpolton Done. So is the idea to put example useage in every method?

Reply:
Maybe. I find from experience that if a method has an example. I am more likely to try it. It provides an alternative route to understand the method and a rapid route to adoption.